clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1451,6 +1456,7 @@ public void commitSync(final Map<TopicPartition, OffsetAndMetadata> offsets, fin
     /**
      * Commit offsets returned on the last {@link #poll(Duration)} for all the subscribed list of topics and partition.
      * Same as {@link #commitAsync(OffsetCommitCallback) commitAsync(null)}
+     * @throws org.apache.kafka.common.errors.FencedInstanceIdException if this consumer instance gets fenced by broker.
```

Comment:
I think commitAsync cannot throw exception directly: only if you registered a callback, then the callback can expose the exception?

Reply:
@guozhangwang Actually it could, when calling `invokeCompletedOffsetCommitCallbacks` it will potentially throw the fenced instance exception.

Reply:
I see. I thought the trace a bit different though: when you call `commitAsync` -> `coordinator.commitOffsetsAsync` -> `invokeCompletedOffsetCommitCallbacks` note that `invokeCompletedOffsetCommitCallbacks` call is just to clear the previous async commit's callback, if there are any, but not related to this commit --- they will be only sent over after that call returns. Which means, if that exception is thrown from `commitAsync`, it would not be for this offset-to-commit, but for a previous call. 

But from user's perspective, it's fenced, it's fenced, no matter if it was triggered by this call or a previous call. So the javadoc looks good.

Reply:
Yes, great summary here. It is trying to populate exception for previous round, so either from `poll` or `commitAsync` calling `invokeCompletedOffsetCommitCallbacks`